baseplate/__init__.py

```diff
@@ -186,10 +186,15 @@ def __init__(
         prefix: Optional[str] = None,
         span: Optional["Span"] = None,
         wrapped: Optional["RequestContext"] = None,
+        kv: Optional[Dict[str, Any]] = None,
```

Comment:
If I'm understanding correctly, it seems like the request context object is kind of already a key-value store, but with statically-defined fields and better typing. I wonder if it would make sense to add the keys you need to the request context object directly instead?

If it's not possible to statically list the keys as fields on this object, maybe we could limit this to just a key-value store for tracing data? I'm a bit worried that a generic key/value store will have more and more keys added to it over the years, and eventually we'd end up in a situation where we never really know what keys are or aren't present at any point in the codebase, or what types the values are.

Reply:
Yeah that makes sense, let me switch things up to at least name it otel specific. OTEL uses a kv store for the context, so it'll be sort of the same thing just named for a specific purpose.

Reply:
This still has the potential to be a bit messy. Generally we are using the global context which is built on top of python ContextVars. This works for most things, but with GQL we're running into an issue with the promise library where execution jobs are put on a queue and are being execute at various times resulting in the context getting screwed up with the global value. Setting the value on the RequestContext object allows us to access it, but for much of the baseplate code we're not using the explicit context value from this object.

I've thought about rewriting the implementation to always use the explicit context value on the the `RequestContext` object but we run into an issue with the `make_object_for_context` function used by the clients. We'd need to pass the context to those calls as well as the "old" server span it currently has as an argument. The problem of updating that function is that it is the interface other clients use to integrate into baseplate.

Either way I can change this to be an otel specific context value, but it should probably only be used in specific scenarios unless we want to rework that make_object_for_context funciton :thinking: